src/NuGet.Clients/NuGet.PackageManagement.UI/Models/PackageDetailControlModel.cs

```diff
@@ -199,7 +210,8 @@ protected override Task CreateVersionsAsync(CancellationToken cancellationToken)
 
                 if (!installed)
                 {
-                    _versions.Add(new DisplayVersion(version.version, additionalInfo: string.Empty, isCurrentInstalled: installed, autoReferenced: autoReferenced, isDeprecated: version.isDeprecated));
+                    VersionRange versionRange = VersionRange.Parse(version.version.ToString(), false);
```

Comment:
Was the problem related to an invalid range? This will just throw, so I'm concerned what UI behavior that will result in.

Reply:
@donnie-msft This change is because for `.config` projects the installation is different and because how DisplayVerison is implemented. `DisplayVersion` always creates a VersionRange property but uses a [different way to create it](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.UI/DisplayVersion.cs#L113). 

And when we do [the installation](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs#L3141) we always create a new version range and not use the from the DisplayVerision class. So for .config files [we write this VersionRange](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/LegacyPackageReferenceProject.cs#L228) to the project files and not [VersionRange.OriginalString](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/CpsPackageReferenceProject.cs#L322).

For example if you wanted to install version 13.0.1 of Newtonsoft.Json in a .config file it will write [13.0.1,13.0.1] and show no updates. This is the behavior of this [bug](https://github.com/NuGet/Home/issues/11425)

Reply:
My point is more around the fact that, this `Parse` will `throw` if it fails. See https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Versioning/VersionRangeFactory.cs#L62

Previous code didn't Parse from what I can tell (quickly glanced at `VersionRange`).
Perhaps that's not possible (pre-validated) and/or we recover from that ?